test/net/sourceforge/kolmafia/maximizer/MaximizerTest.java

```diff
@@ -1225,6 +1225,43 @@ public void shouldSuggestReplicaParka() {
         assertTrue(someBoostIs(x -> commandStartsWith(x, "parka ghostasaurus")));
       }
     }
+
+    @Nested
+    class GarbageTote {
+      @Test
+      public void shouldSuggestEquippingGarbageToteItem() {
+        final var cleanups =
+            new Cleanups(withItem(ItemPool.GARBAGE_TOTE), withItem(ItemPool.BROKEN_CHAMPAGNE));
+
+        try (cleanups) {
+          assertTrue(maximize("weapon damage percent"));
+          recommendedSlotIs(Slot.WEAPON, "broken champagne bottle");
+          assertTrue(someBoostIs(x -> commandStartsWith(x, "fold ¶9692;equip weapon ¶9692")));
```

Comment:
Should we recommend folding an item if we already have it? Did it do that before your change?

Reply:
I wrote this test before making the change.

I originally wrote it to assume it was not going to recommend folding, but when I put in debug printing to see what all the boosts were, I was surprised to see a fold in there.

With the change, with tinsel tights in inventory, looking at the Maximizer frame:

item drop:
fold and equip weapon broken champagne bottle (+110)

monster level:
equip pants tinsel tights (+25)

Reply:
Look at this.
```
      @Test
      public void shouldSuggestEquippingGarbageToteItem1() {
        final var cleanups =
            new Cleanups(withItem(ItemPool.GARBAGE_TOTE), withItem(ItemPool.TINSEL_TIGHTS));

        try (cleanups) {
          assertTrue(maximize("monster level"));
          recommendedSlotIs(Slot.PANTS, "tinsel tights");
          for (var boost : getBoosts()) {
            System.out.println(boost.getCmd());
          }
          assertTrue(someBoostIs(x -> commandStartsWith(x, "equip pants ¶9693")));
        }
      }

      @Test
      public void shouldSuggestEquippingGarbageToteItem2() {
        final var cleanups =
            new Cleanups(withItem(ItemPool.GARBAGE_TOTE), withItem(ItemPool.BROKEN_CHAMPAGNE));

        try (cleanups) {
          assertTrue(maximize("weapon damage percent"));
          recommendedSlotIs(Slot.WEAPON, "broken champagne bottle");
          for (var boost : getBoosts()) {
            System.out.println(boost.getCmd());
          }
          assertTrue(someBoostIs(x -> commandStartsWith(x, "fold ¶9692;equip weapon ¶9692")));
        }
      }
```
yields this:
```
MaximizerTest > Modeables > GarbageTote > shouldSuggestEquippingGarbageToteItem1() STANDARD_OUT
    equip pants ¶9693

MaximizerTest > Modeables > GarbageTote > shouldSuggestEquippingGarbageToteItem2() STANDARD_OUT
    fold ¶9692;equip weapon ¶9692
```

On my character that currently has tinsel tights:

weapon damage percent:
equip weapon replica haiku katana (+50)
fold and equip off-hand broken champagne bottle (+50)

I executed that and now it says:

weapon damage percent:
keep weapon replica haiku katana (+50)
keep off-hand broken champagne bottle (+50)

I dropped the broken champaign bottle and it now says this:

weapon damage percent:
keep weapon replica haiku katana (+50)
equip off-hand broken champagne bottle (+50)

It sure seems like it is doing the right thing. I have no idea why the "maximize" command is behaving like that in the test.

Reply:
One more test:
```
      @Test
      public void shouldSuggestEquippingGarbageToteItem2() {
        final var cleanups =
            new Cleanups(
                withItem(ItemPool.REPLICA_GARBAGE_TOTE),
                withItem(ItemPool.REPLICA_HAIKU_KATANA),
                withItem(ItemPool.BROKEN_CHAMPAGNE),
                withSkill("Double-Fisted Skull Smashing"));

        try (cleanups) {
          assertTrue(maximize("weapon damage percent"));
          recommendedSlotIs(Slot.OFFHAND, "broken champagne bottle");
          for (var boost : getBoosts()) {
            System.out.println(boost.getCmd());
          }
          assertTrue(someBoostIs(x -> commandStartsWith(x, "fold ¶9692;equip off-hand ¶9692")));
        }
      }
```
yields:
```
MaximizerTest > Modeables > GarbageTote > shouldSuggestEquippingGarbageToteItem2() STANDARD_OUT
    equip weapon ¶11203
    fold ¶9692;equip off-hand ¶9692
```

As I said, the maximizer frame is peachy.
I have no idea why the maximizer function is being weird.

I'll submit the second test - one for tinsel tights, one for broken champagne bottle - in case we want to investigate later, but will shrug and merge as is.

Reply:
My thoughts were that normally, folding would do nothing for an item we already have, but for the champagne bottle it would reset the charges.

So long as that's how it worked, it's fine for it to continue to work this way.

Reply:
Look at this code in Maximizer.java:
```
      // We might want to fold for a new Garbage item, even if we already have it, to reset it
      if ((itemId == ItemPool.BROKEN_CHAMPAGNE
              && Preferences.getInteger("garbageChampagneCharge") == 0)
          || (itemId == ItemPool.MAKESHIFT_GARBAGE_SHIRT
                  && Preferences.getInteger("garbageShirtCharge") == 0)
              && !Preferences.getBoolean("_garbageItemChanged")) {
        if (checkedItem.initial > count) {
          text = "fold & " + text;
          cmd = "fold \u00B6" + item.getItemId() + ";" + cmd;
        }
```
Given operator precedence, that is A || (B && C) when I bet it is intended to be (A || B) && C.

Reply:
That is correct: if the champagne bottle has zero charges left, it will fold it again. That will only recharge it if you last folded the champagne bottle yesterday - I think - but, as coded, we can only detect that you folded SOMETHING today.

As coded, it will not re-fold if you have charges left. I just opened another PR with a test for that.